src/components/Button/index.js

```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+import { Container, TextButton } from './styles';
+
+export const Button = ({ text, ...props }) => {
```

Comment:
Suas props devem ser tinadas e seu componente tem que indicar qual o retorno dele. Pode criar uma interface Props nesse arquivo mesmo.
```suggestion
export const Button = ({ text, ...props }: Props): ReactElement => {
```